### PR TITLE
jira/HZN-376: JMX Config Generator Documentation

### PR DESCRIPTION
**Pull request by christianpape**
Please review and merge!

Issue: http://issues.opennms.org/browse/HZN-376
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS104

### CODE REVIEW

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 1]
```diff
@@ -1,59 +1,252 @@
...
-:imagesdir: ../../images
+The _JMX Config Generator_ is a revised and improvised version of the original _JmxConfigTool_.
+It provides a fast and easy way to generate a valid and functional `jmx-datacollection-config.xml` file from a running Java application.
+Based on this generated configuration file it can also generate a `snmp-graph.properties` file with matching graph definitions.
 
-=== JmxConfigTool
-The JmxConfigTool provides a fast and easy way to generate a proper _jmx-datacollection-config_ from a running Java application.
-Based on the generated configuration file it can also provide a set of _snmp-graph_ configurations.
+==== Installation
+From current _OpenNMS_ versions 1.12 and later the _JMX Config Generator_ is included as a separate installable package.
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: Sine _OpenNMS_ version 1.12 the _JMX Config Generator_ is included as a separate installable package.

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 2]
```diff
@@ -1,59 +1,252 @@
...
+Need to get 2,180 kB of archives.
+After this operation, 2,225 kB of additional disk space will be used.
+Get:1 http://debian.opennms.eu/ stable/main opennms-jmx-config-generator all 16.0.2-1 [2,180 kB]
+Fetched 2,180 kB in 0s (3,275 kB/s)
+Selecting previously unselected package opennms-jmx-config-generator.
+(Reading database ... 153349 files and directories currently installed.)
+Preparing to unpack .../opennms-jmx-config-generator_16.0.2-1_all.deb ...
+Unpacking opennms-jmx-config-generator (16.0.2-1) ...
+Setting up opennms-jmx-config-generator (16.0.2-1) ...
+----
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: Do we have to add a note that you have to add the opennms repository first?

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 3]
```diff
@@ -1,59 +1,252 @@
...
+Based on this generated configuration file it can also generate a `snmp-graph.properties` file with matching graph definitions.
 
-=== JmxConfigTool
-The JmxConfigTool provides a fast and easy way to generate a proper _jmx-datacollection-config_ from a running Java application.
-Based on the generated configuration file it can also provide a set of _snmp-graph_ configurations.
+==== Installation
+From current _OpenNMS_ versions 1.12 and later the _JMX Config Generator_ is included as a separate installable package.
 
-The JmxConfigTool offers two modes:
+===== Installation on Ubuntu
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: `Ubuntu/Debian`?

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 4]
```diff
@@ -1,59 +1,252 @@
...
+
+Complete!
+
 ----
 
-===== Credentials
+===== Installation from Source
+The _OpenNMS_ source is hosted on http://github.com/OpenNMS/opennms[GitHub].
+Please refer to the Wiki page http://www.opennms.org/wiki/Developing_with_Git#Getting_OpenNMS[Developing with Git] for more information.
+After cloning the repository you have to enter enter the folder `opennms/features/opennms-jmx-tools`.
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: the folder name should be `opennms/features/jmx-config-generator`

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 5]
```diff
@@ -1,59 +1,252 @@
...
+Next, invoke `mvn package` to build the _JMX Config Generator_.
+Inside the newly created `target` folder a file named like `jmxconfiggenerator-<VERSION>-onejar.jar` is present.
+This file can be invoked by:
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: please remove the "like" in the sentence.

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 6]
```diff
@@ -89,35 +282,57 @@ CommittedVirtualMemorySize CommiVirtuMemorSize
...
 
 [source, shell]
 ----
-$ java -jar opennms_jmx_config_generator.jar -graph -input myJmxConfig.xml -out myJmxSnmpGraphs.properties -template myTemplate.vm
+$ jmx-config-generator generate-graph --input myconfig.xml --output mygraph.properties --template mytemplate.vm
 ----
 
-The template file has to be an Apache Velocity template.
-The sample represents the template that is used by default:
+The template file has to be an _Apache Velocity_ template.
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: Shall we add a link to the apache velocity project page?

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc [hunk 7]
```diff
@@ -89,35 +282,57 @@ CommittedVirtualMemorySize CommiVirtuMemorSize
...
+
+===== Command line options
+
+[options="header, autowidth"]
+|===
+| Option/Argument                    | Description                                                                                        | Default
+| `--input <jmx-datacollection.xml>` | Configuration file to use as input to generate the graph properties file                           | -
+| `--output <file>`                  | Output filename for the generated graph properties file.                                           | -
+| `--print-template`                 | Prints the default template.                                                                       | false
+| `--template <file>`                | Template file using _Apache Velocity_ template engine to be used to generate the graph properties. | -
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: Shall we add a link to the apache velocity project page?

**Comment by mvrueden** [comment]
Besides my small nitty picky comments, this issue looks good. @christianpape can you please have a look at my notes and let me know what you think :)

File: opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc (hunk 2 above)
Comment (christianpape) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: It's mentioned above: "Install the package from the _OpenNMS_ repository". So, I think this is okay.

Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/jmx-config-generator.adoc]: That is maybe the case, but no-where in the admin guide is described/explained what that means. Can we at least add a note or a link to where that is explained?

**Comment by christianpape** [comment]
I fixed the rest of the things you mentioned. Jay!

**Comment by mvrueden** [comment]
:+1:

**Pull request merged**